fdb-record-layer-lucene/README.md

```diff
@@ -0,0 +1,14 @@
+# FoundationDB Record Layer Lucene Index
+
+The lucene index implementation on the record layer consists of two key components.  The first component is a Directory implementation backed by an FDB KeySpace.
+
+* **Directory Implementation** 
+
+A Directory provides an abstraction layer for storing a list of files. A directory contains only files (no sub-folder hierarchy). Implementing classes must comply with the following:
+A file in a directory can be created (createOutput(java.lang.String, org.apache.lucene.store.IOContext)), appended to, then closed.
+A file open for writing may not be available for read access until the corresponding IndexOutput is closed.
+Once a file is created it must only be opened for input (openInput(java.lang.String, org.apache.lucene.store.IOContext)), or deleted (deleteFile(java.lang.String)). Calling createOutput(java.lang.String, org.apache.lucene.store.IOContext) on an existing file must throw FileAlreadyExistsException.
+
```

Comment:
Yes, that is exactly what Lucene does under the cover via this Directory API.  You can set a max capacity for the Segment (shards) size and that will then govern how many segments you have to linearly search for.  Let me add some more comments to the class to outline this to answer your question.

Reply:
So does that mean we would have two different levels of compacting data for text? In FDB (at least in Rocks, so FDB +1 (LSM)) we would have a compaction going on inside the database but it looks like here we would have, too, compaction going on on lucene block level, right? That seems that this would create an amount of client based churn and churn in general. Sure I am missing something.
The second thing here is the serialization to bytes and back. It feels like the abstraction here is too low. I get that's given by the interfaces on the Lucene side, but what's happening here is that these blocks of bytes are completely opaque to fdb. You will have to fetch/cache a block in order to have lucene search for something in it.

Reply:
That is correct for the multiple level of compaction.  Sorry, this is the **abstraction** for Lucene but I understand your concerns.